raddb/sites-available/ldap_sync

```diff
@@ -27,194 +102,391 @@
+			#  specified in this configuration.  The key limitation is
+			#  <=, >= and bitwise filters assume attributes are numeric.
+			#
+			#  The only extensible match filters supported are the Active
+			#  Directory bitwise AND and OR.
+			#
+			#  A suitable filter, to only receive notificaitons regarding
+			#  normal user accounts could be:
+			#
+			#    (userAccountControl:1.2.840.113556.1.4.803:=512)
+			#
+			#  In addition, there is nothing returned by Active Directory to
+			#  distinguish between an object being added or being modified.
+			#  All LDAP entries which are returned are therefore processed
+			#  through the recv Modify section when the directory is Active
+			#  Directory.
+			#
+			#  By default Active Directory puts a limit of 5 on the number
+			#  of persistent searches which can be active on a connection.
+			#
+			#  To determine if an object is enabled or disabled, the attribute
+			#  userAccountControl can be evaluated.  This is returned as
+			#  string data, so will want mapping to an integer attribute
+			#  for processing.
+			#
+			#  Firstly, one based on a naming context which covers all
+			#  user objects.  This will return results when objects are
+			#  added, modified or restored from the Deleted Objects
+			#  container.
+			#
+#			sync {
+#				base_dn = 'cn=Users,dc=example,dc=com'
+#				filter = '(userAccountControl:1.2.840.113556.1.4.803:=512)'
+#				scope = 'sub'
+#				attrs = 'userAccountControl'
+#				attrs = 'sAMAccountName'
+#
+#				update {
+#					&User-Name := 'sAMAccountName'
+#					&Tmp-Integer-0 := 'userAccountControl'
+#				}
+#			}
 
-			#  Specify the list of attributes to return in entries,
-			#  one config item per LDAP attribute.
 			#
-			#  If no attr config items are specified, then all attributes
-			#  will be returned.
-#			attr = 'cn'
-#			attr = 'foo'
+			#  Secondly, if you have the Recycle Bin enabled in Active
+			#  Directory and wish to be notified about deleted objects,
+			#  add a search covering the Deleted Objects container.
+			#
+			#  This will return results when an object is deleted, however
+			#  the DN and CN of the object are changed.  The attribute
+			#  lastKnownParent identifies where the object was deleted
+			#  from.  However, lastKnownParent may not be returned when
+			#  searching the Global Catalog.
+			#
+			#  Ensure this sync also includes the attribute isDeleted which
+			#  is used to direct processing of packets received to the
+			#  recv Delete section.
+			#
+#			sync {
+#				base_dn = "CN=Deleted Objects,dc=example,dc=com"
+#				filter = '(userAccountControl:1.2.840.113556.1.4.803:=512)'
+#				scope = "one"
+#				attrs = 'isDeleted'
+#				attrs = 'sAMAccountName'
+#				attrs = 'userAccountControl'
+#				attrs = 'lastKnownParent'
+#
+#				update {
+#					&User-Name := 'sAMAccountName'
+#					&Tmp-Integer-0 := 'userAccountControl'
+#					&Tmp-String-0 := 'lastKnownParent'
+#				}
```

Comment:
Is there a reason to have the attributes manually specified, or could we just take them from the RHS of the update section?